python-ecosys/urequests/urequests.py

```diff
@@ -7,6 +7,17 @@ def __init__(self, f):
         self.encoding = "utf-8"
         self._cached = None
 
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        data = b""
+        chunk = self.raw.read()
+        while chunk:
+            data += chunk
+            chunk = self.raw.read()
+        return data
```

Comment:
Does it make sense to combine all data here?  Why not just return the result of the first read?  The caller must anyway keep calling this function until it has all data.

Reply:
Good point, that makes sense and the loop just adds an unnecessary read. I will test it with `return self.raw.read()' and report back.